src/main/java/pl/edu/mimuw/FormulaFactory.java

```diff
@@ -0,0 +1,43 @@
+package pl.edu.mimuw;
+
+import java.util.HashMap;
+
+public class FormulaFactory {
+  protected static HashMap<String, Boolean> values;
```

Comment:
nie mieszajmy `static` z tworzeniem instancji dla takich klas, które są jedynie fabrykami obiektów. Jeżeli możesz, to przypomnij mi o tym na zajęciach, to pogadamy, jak to powinno być, bo tu jest troszkę namieszane i długo by 😉